### PR TITLE
Geiger counters can now scan items

### PR DESCRIPTION
**Pull request by silicons**
Why not?

### CODE REVIEW

**Comment by JamesDenholm** [comment]
http://puu.sh/AgE8m/6a71610939.png
This might be why not

**Comment by kevinz000** [comment]
Fuck.

File: code/game/objects/items/devices/geiger_counter.dm [hunk 1]
```diff
@@ -147,7 +147,7 @@
 			var/total = 0
 			for(var/i in get_rad_contents(target))
 				var/atom/scanning = i
-				var/datum/component/radioactive/R = GET_COMPONENT_FROM(/datum/component/radioactive, scanning)
+				var/datum/component/radioactive/R = scanning.GetComponent(/datum/component/radioactive)
```
Comment (Barhandar) [inline code/game/objects/items/devices/geiger_counter.dm]: GET_COMPONENT_FROM(R,/datum/component/radioactive, scanning)

Whole line, not just the right side.

Comment (kevinz000) [reply code/game/objects/items/devices/geiger_counter.dm]: i know but it's the same thing

File: code/game/objects/items/devices/geiger_counter.dm [hunk 2]
```diff
@@ -131,17 +131,28 @@
...
-			radiation_count = 0
-		return 1
-	..()
+			var/total = 0
+			for(var/i in get_rad_contents(target))
+				var/atom/scanning = i
+				var/datum/component/radioactive/R = scanning.GetComponent(/datum/component/radioactive)
+				if(R)
+					total += R.strength
+			to_chat(user, "<span class='warning'>You scan [target] with [src], and it reports a radiation count of [total].</span>")
```
Comment (ninjanomnom) [inline code/game/objects/items/devices/geiger_counter.dm]: You need to be using the scan proc

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/geiger_counter.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/devices/geiger_counter.dm]: what's the benefit @Barhandar ?

**Review by ninjanomnom — APPROVED** [review]

File: code/game/objects/items/devices/geiger_counter.dm (hunk 1 above)
Comment (Barhandar) [reply code/game/objects/items/devices/geiger_counter.dm]: ask Cyberboss

**Pull request merged**